draft-ietf-quic-transport.md

```diff
@@ -2000,2812 +2295,2593 @@ codepoints in subsequent packets. Doing so allows the connection to traverse
+|:------------|:---------------------|:-------------------------------|
+| 0x00        | PADDING              | {{frame-padding}}              |
+| 0x01        | RST_STREAM           | {{frame-rst-stream}}           |
+| 0x02        | CONNECTION_CLOSE     | {{frame-connection-close}}     |
+| 0x03        | APPLICATION_CLOSE    | {{frame-application-close}}    |
+| 0x04        | MAX_DATA             | {{frame-max-data}}             |
+| 0x05        | MAX_STREAM_DATA      | {{frame-max-stream-data}}      |
+| 0x06        | MAX_STREAM_ID        | {{frame-max-stream-id}}        |
+| 0x07        | PING                 | {{frame-ping}}                 |
+| 0x08        | BLOCKED              | {{frame-blocked}}              |
+| 0x09        | STREAM_BLOCKED       | {{frame-stream-blocked}}       |
+| 0x0a        | STREAM_ID_BLOCKED    | {{frame-stream-id-blocked}}    |
+| 0x0b        | NEW_CONNECTION_ID    | {{frame-new-connection-id}}    |
+| 0x0c        | STOP_SENDING         | {{frame-stop-sending}}         |
+| 0x0d        | RETIRE_CONNECTION_ID | {{frame-retire-connection-id}} |
+| 0x0e        | PATH_CHALLENGE       | {{frame-path-challenge}}       |
+| 0x0f        | PATH_RESPONSE        | {{frame-path-response}}        |
+| 0x10 - 0x17 | STREAM               | {{frame-stream}}               |
+| 0x18        | CRYPTO               | {{frame-crypto}}               |
+| 0x19        | NEW_TOKEN            | {{frame-new-token}}            |
+| 0x1a - 0x1b | ACK                  | {{frame-ack}}                  |
+{: #frame-types title="Frame Types"}
 
+All QUIC frames are idempotent.  That is, a valid frame does not cause
+undesirable side effects or errors when received more than once.
 
-### Immediate Close
+The Frame Type field uses a variable length integer encoding (see
+{{integer-encoding}}) with one exception.  To ensure simple and efficient
+implementations of frame parsing, a frame type MUST use the shortest possible
+encoding.  Though a two-, four- or eight-octet encoding of the frame types
+defined in this document is possible, the Frame Type field for these frames is
+encoded on a single octet.  For instance, though 0x4007 is a legitimate
+two-octet encoding for a variable-length integer with a value of 7, PING frames
+are always encoded as a single octet with the value 0x07.  An endpoint MUST
+treat the receipt of a frame type that uses a longer encoding than necessary as
+a connection error of type PROTOCOL_VIOLATION.
 
-An endpoint sends a closing frame (CONNECTION_CLOSE or APPLICATION_CLOSE) to
-terminate the connection immediately.  Any closing frame causes all streams to
-immediately become closed; open streams can be assumed to be implicitly reset.
 
-After sending a closing frame, endpoints immediately enter the closing state.
-During the closing period, an endpoint that sends a closing frame SHOULD respond
-to any packet that it receives with another packet containing a closing frame.
```

Comment:
I'm not sure that I like the fact that the specific transport parameters are described way back up top now.  I think that having the description of each down here is better.  There's a lot of encoding information in them, and we don't generally rely on the definitions of transport parameters for much.  Anything we do need can use a forward reference.